src/core/config/transformer/preset.rs

```diff
@@ -26,6 +27,7 @@ impl Transform for Preset {
             .pipe(super::TreeShake.when(self.tree_shake))
             .pipe(super::TypeMerger::new(self.merge_type))
             .pipe(super::ImproveTypeNames.when(self.use_better_names))
+            .pipe(super::FlattenSingleTypes.when(self.flatten_single_types))
```

Comment:
Issue wa asking to put the flatten of single types before ImproveTypeNames but i noticed some strange behaviour when it is before so i placed it after.